homeassistant/components/alarm_control_panel/mqtt.py

```diff
@@ -49,44 +50,52 @@
     vol.Optional(CONF_PAYLOAD_DISARM, default=DEFAULT_DISARM): cv.string,
     vol.Optional(CONF_UNIQUE_ID): cv.string,
     vol.Optional(CONF_DEVICE): mqtt.MQTT_ENTITY_DEVICE_INFO_SCHEMA,
+    vol.Optional(CONF_VALUE_TEMPLATE): cv.template,
+    vol.Optional(CONF_CODE_ARM_REQUIRED, default=True): cv.boolean,
 }).extend(mqtt.MQTT_AVAILABILITY_SCHEMA.schema)
 
 
 async def async_setup_platform(hass: HomeAssistantType, config: ConfigType,
                                async_add_entities, discovery_info=None):
+    value_template = config.get(CONF_VALUE_TEMPLATE)
```

Comment:
Don't parse the config here, it breaks discovery update.
Do it like it's done in `switch.mqtt` instead: https://github.com/home-assistant/home-assistant/blob/b80bed64f5087a15d54e58b1ed2328baf20c865d/homeassistant/components/switch/mqtt.py#L137